SNIPS/snip-3668.md

```diff
@@ -0,0 +1,75 @@
+---
+snip: 3668
+title: Starknet CCIP Read
+author: Elias Tazartes <@Eikix>
+status: Draft
+type: SRC
+created: 2024-08-17
+---
+
+## Simple Summary
+
+Implementation of the CCIP-Read standard ([ERC-3668](https://eips.ethereum.org/EIPS/eip-3668)) specifically for the Starknet state, making it possible for L1 contracts (and later any L2 contracts) to fetch contract state data from Starknet.
```

Comment:
what do you mean by "later any L2 contracts"?

Reply:
like a Solidity contract on Arbitrum should be able to read the state of Starknet by:
- doing a L1SLOAD call to get the current state root of Starknet core contract
- verifying a merkle inclusion proof of a slot against that state root